infrastructure/authentication/auth_service_impl.go

```diff
@@ -68,14 +69,44 @@ func (a *AuthenticationServiceImpl) Provider() AuthenticationProvider {
 
 func (a *AuthenticationServiceImpl) Authenticate(ctx context.Context) (token string, err error) {
 	token, err = a.provider.Authenticate(ctx)
+
 	if token == "" || err != nil {
 		a.c.Logger().Warn().Err(err).Msgf("Failed to authenticate using auth provider %v", reflect.TypeOf(a.provider))
 		return token, err
 	}
+
+	customUrl := a.c.SnykApi()                                                    //returns https://api.snyk.io/v1
+	engineUrl := a.c.Engine().GetConfiguration().GetString(configuration.API_URL) //Get url from GAF, https://api.snyk.io
+	prioritizedUrl := getPrioritizedApiUrl(customUrl, engineUrl)
+
+	a.c.UpdateApiEndpoints(prioritizedUrl)
 	a.UpdateCredentials(token, true)
+	a.ConfigureProviders(a.c)
+
 	return token, err
 }
 
+func getPrioritizedApiUrl(customUrl string, engineUrl string) string {
+	defaultUrl := config.DefaultSnykApiUrl
+
+	// If the custom URL is not changed (equals default) and no engine URL is provided,
+	// use the default URL.
+	if customUrl == defaultUrl && engineUrl == "" {
+		return defaultUrl
+	}
+
+	// If the custom URL equals the default but an engine URL is provided, use the engine URL.
+	// The authentication flow has redirected the user to the correct endpoint.
+	// http://api.eu.snyk.io
+	if customUrl == defaultUrl {
```

Comment:
clean custom Url before comparison
just in case an extra space or a trailing slash is there

Reply:
u can add the notification here